admin/src/main/java/edu/unc/lib/dl/admin/controller/ResubmitIngestController.java

```diff
@@ -0,0 +1,176 @@
+import edu.unc.lib.dl.util.RedisWorkerConstants.DepositState;
+import edu.unc.lib.dl.util.MetsHeaderScanner;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.util.DepositConstants;
+import edu.unc.lib.dl.util.DepositStatusFactory;
+
+@Controller
+public class ResubmitIngestController {
+	
+	private static final Logger log = LoggerFactory.getLogger(ResubmitIngestController.class);
+	
+	@Autowired
+	private DepositStatusFactory depositStatusFactory;
+	
+	@Autowired
+	private File depositsDirectory;
+	
+	@RequestMapping(value = "ingest/{pid}/resubmit", method = RequestMethod.POST)
+	public @ResponseBody
+	Map<String, ? extends Object> handleResubmit(@PathVariable("pid") String pid, @RequestParam("file") MultipartFile uploadedFile, HttpServletRequest request, HttpServletResponse response) {
+		
+		Map<String, Object> result = new HashMap<String, Object>();
+		PID depositPid = new PID(pid);
+		
+		
+		// Try to retrieve the deposit status
+		
+		Map<String, String> status = depositStatusFactory.get(depositPid.getUUID());
+		
+		if (status == null || status.isEmpty()) {
+			response.setStatus(400);
+			result.put("error", "Couldn't find the deposit.");
+			return result;
+		}
+		
+		
+		// Check that the current user is either an admin or was the depositor
+		
+		AccessGroupSet groups = GroupsThreadStore.getGroups();
+		String depositorName = status.get(DepositField.depositorName.name());
+		
+		if (!(groups.contains(AccessGroupConstants.ADMIN_GROUP) || depositorName.equals(GroupsThreadStore.getUsername()))) {
+			response.setStatus(401);
+			return result;
+		}
+		
+		
+		// Check that the deposit is in the failed state
+		
+		String state = status.get(DepositField.state.name());
+		
+		if (state == null || !state.equals(DepositState.failed.name())) {
+			response.setStatus(400);
+			result.put("error", "The deposit isn't in the failed state.");
+			return result;
+		}
+		
+		
+		// Create the resubmit deposit directory
+		
+		String resubmitDirName;
+		
+		try {
+			resubmitDirName = createResubmitDirectory();
+		} catch (IOException e) {
+			response.setStatus(500);
+			result.put("error", "Couldn't save the resubmitted ingest package.");
+			return result;
+		}
+		
+		File resubmitDir = new File(depositsDirectory, resubmitDirName);
+		File resubmitDataDir = new File(resubmitDir, DepositConstants.DATA_DIR);
+		
+		
+		// Transfer the uploaded file to the resubmit data directory
+		
+		File resubmitPackageFile = new File(resubmitDataDir, uploadedFile.getOriginalFilename());
+		
+		try {
+			uploadedFile.transferTo(resubmitPackageFile);
+		} catch (IOException e) {
+			response.setStatus(500);
+			result.put("error", "Couldn't save the resubmitted ingest package.");
+			return result;
+		}
+		
+		
+		// Scan the ingest file
+		
+		MetsHeaderScanner scanner = new MetsHeaderScanner();
+		
+		try {
+			scanner.scan(resubmitPackageFile, uploadedFile.getOriginalFilename());
+		} catch (Exception e) {
+			response.setStatus(500);
```

Comment:
I'm glad that it is giving less crazy error messages to the client, but we probably do need to have some way to get to the stacktrace of why the ingest failed.  That could just be dumping a stacktrack through log4j